src/Compilers/Core/Portable/SourceGeneration/GeneratorAdaptor.cs

```diff
@@ -15,6 +15,11 @@ namespace Microsoft.CodeAnalysis
     /// </summary>
     internal sealed class SourceGeneratorAdaptor : IIncrementalGenerator
     {
+        /// <summary>
+        /// A dummy extension that is used to indicate this adaptor was created outside of the driver.
+        /// </summary>
+        public const string DummySourceExtension = ".dummy";
+
         private readonly string _sourceExtension;
```

Comment:
Is this a value that is controllable by customers? If so have reservations about using a magic string. Had trouble tracking this through all the APIs to see where it originates.

Reply:
No, it's only created internally. We track the file extension because we actively parse the tree, so we need to pass it in from the outer driver when we wrap an ISourceGenerator. 

The generators on this path are never actually used directly, so we can just set it to anything. I chose a magic string so we can assert later on that it *isn't* used and dings us if we ever do allow it, so we know to fix it. 

At some point we should probably refactor these so the string isn't needed at all, but this just seemed simpler for now.